nixos/modules/services/networking/networkmanager.nix

```diff
@@ -379,6 +380,52 @@ in {
           for more details.
         '';
       };
+      ensure-profiles = {
+        profiles = mkOption {
+          type = types.attrsOf ini.type;
+          default = { };
+          example = {
+            home-wifi = {
+              connection = {
+                id = "home-wifi";
+                type = "wifi";
+                permissions = "";
+              };
+              wifi = {
+                mac-address-blacklist = "";
+                mode = "infrastructure";
+                ssid = "Home Wi-Fi";
+              };
+              wifi-security = {
+                auth-alg = "open";
+                key-mgmt = "wpa-psk";
+                psk = "$PASSWORD";
+              };
+              ipv4 = {
+                dns-search = "";
+                method = "auto";
+              };
+              ipv6 = {
+                addr-gen-mode = "stable-privacy";
+                dns-search = "";
+                method = "auto";
+              };
+            };
+          };
+          description = lib.mdDoc ''
+            declaratively define NetworkManager profiles.
+          '';
+        };
+        environmentFiles = mkOption {
+          default = [];
+          type = types.listOf types.path;
+          example = [ "/run/secrets/network-manager.env" ];
+          description = lib.mdDoc ''
+            Files to load as environment file. Environment variables from this file
+            will be substituted into the static configuration file using envsubst.
```

Comment:
I honestly never heard of that specific tool maybe you can also add a docs link? (Definitely only a nice to have.)

Reply:
Sure, I just followed @999eagle 's example in https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/web-servers/traefik.nix